src/librustc_error_codes/lib.rs

```diff
@@ -0,0 +1,26 @@
+//! This library is used to gather all error codes into one place. The goal
+//! being to make their maintenance easier.
+
+#[macro_export]
+macro_rules! register_diagnostics {
```

Comment:
And then this macro can be inlined to where it is defined.

Reply:
Again: I prefer to keep them separated, makes the reading simpler.